admin/services/pupil-census.service.js

```diff
@@ -38,45 +34,55 @@ pupilCensusService.upload = async (uploadFile) => {
         }
       })
   })
-  const blobResult = await pupilCensusService.uploadToBlobStorage(csvData)
-  await pupilCensusService.create(uploadFile, blobResult)
-}
-/**
- * Upload stream to Blob Storage
- * @param uploadFile
- * @return {Promise<void>}
- */
-pupilCensusService.uploadToBlobStorage = async (uploadFile) => {
-  const streamLength = pupilCensusMaxSizeFileUploadMb
-  const remoteFilename = `${uuidv4()}_${moment().format('YYYYMMDDHHmmss')}.csv`
-  const csvFileStream = uploadFile.join('\n')
-  return azureFileDataService.azureUploadFile('censusupload', remoteFilename, csvFileStream, streamLength)
+  // Remove headers from csv
+  csvData.shift()
+  // Create the pupil census record
+  const job = await pupilCensusService.create(uploadFile)
+  if (!job || !job.insertId) {
+    throw new Error('Job has not been created')
+  }
+  // Process and perform pupil bulk insertion
+  const submissionResult = await pupilCensusProcessingService.process(csvData, job.insertId)
+  if (!submissionResult) {
+    throw new Error('No result has been returned from pupil bulk insertion')
+  }
+  // Update pupil census record with corresponding output
+  pupilCensusService.updateJobOutput(job.insertId, submissionResult)
 }
 
 /**
  * Creates a new pupilCensus record
  * @param {Object} uploadFile
- * @param {Object} blobResult
- * @return {Object}
+ * @return {Promise}
  */
-pupilCensusService.create = async (uploadFile, blobResult) => {
-  let dataInput = []
+pupilCensusService.create = async (uploadFile) => {
   const csvName = uploadFile.filename && uploadFile.filename.replace(/\.[^/.]+$/, '')
-  const blobFileName = blobResult && blobResult.name
-  dataInput.push(csvName, blobFileName)
-  dataInput = JSON.stringify(dataInput.join(','))
   const jobType = await jobTypeDataService.sqlFindOneByTypeCode('CEN')
   const jobStatus = await jobStatusDataService.sqlFindOneByTypeCode('SUB')
   const pupilCensusRecord = {
-    jobInput: dataInput,
+    jobInput: csvName,
     jobType_id: jobType.id,
     jobStatus_id: jobStatus.id
   }
-  await jobDataService.sqlCreate(pupilCensusRecord)
+  return jobDataService.sqlCreate(pupilCensusRecord)
+}
+
+/**
+ * Updates the output of a pupilCensus record
+ * @param {Number} jobId
+ * @param {Object} submissionResult
+ * @returns {Promise.<void>}
+ */
+pupilCensusService.updateJobOutput = async (jobId, submissionResult) => {
+  const jobStatusCode = submissionResult.errorOutput ? 'CWR' : 'COM'
```

Comment:
magic strings

Reply:
@jon-shipley this is the job status codes we are using in the job status table. 
CWR = Completed with Errors
COM = Completed

Reply:
Please replace with constants